presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -0,0 +1,188 @@
+            case IS_DISTINCT_FROM:
+                break;
+        }
+
+        return filterStatsForUnknownExpression(inputStatistics);
+    }
+
+    private PlanNodeStatsEstimate symbolToLiteralGreaterThan(Symbol symbol, double literal)
+    {
+        SymbolStatsEstimate symbolStats = inputStatistics.getSymbolStatistics(symbol);
+
+        StatisticRange range = new StatisticRange(symbolStats.getLowValue(), symbolStats.getHighValue(), symbolStats.getDistinctValuesCount());
+        StatisticRange intersectRange = range.intersect(new StatisticRange(literal, POSITIVE_INFINITY, NaN));
+
+        double filterFactor = range.overlapPercentWith(intersectRange);
+        SymbolStatsEstimate symbolNewEstimate =
+                SymbolStatsEstimate.builder()
+                        .setAverageRowSize(symbolStats.getAverageRowSize())
+                        .setDistinctValuesCount(intersectRange.getDistinctValuesCount())
+                        .setHighValue(intersectRange.getHigh())
+                        .setLowValue(intersectRange.getLow())
+                        .setNullsFraction(0.0).build();
+
+        return inputStatistics.mapOutputRowCount(rowCount -> filterFactor * (1 - symbolStats.getNullsFraction()) * rowCount)
+                .mapSymbolColumnStatistics(symbol, oldStats -> symbolNewEstimate);
+    }
+
+    private PlanNodeStatsEstimate symbolToLiteralLessThan(Symbol symbol, double literal)
+    {
+        SymbolStatsEstimate symbolStats = inputStatistics.getSymbolStatistics(symbol);
+
+        StatisticRange range = new StatisticRange(symbolStats.getLowValue(), symbolStats.getHighValue(), symbolStats.getDistinctValuesCount());
+        StatisticRange intersectRange = range.intersect(new StatisticRange(NEGATIVE_INFINITY, literal, NaN));
+
+        double filterFactor = range.overlapPercentWith(intersectRange);
+        SymbolStatsEstimate symbolNewEstimate =
+                SymbolStatsEstimate.builder()
+                        .setAverageRowSize(symbolStats.getAverageRowSize())
+                        .setDistinctValuesCount(intersectRange.getDistinctValuesCount())
+                        .setHighValue(intersectRange.getHigh())
+                        .setLowValue(intersectRange.getLow())
+                        .setNullsFraction(0.0).build();
+
+        return inputStatistics.mapOutputRowCount(rowCount -> filterFactor * (1 - symbolStats.getNullsFraction()) * rowCount)
+                .mapSymbolColumnStatistics(symbol, oldStats -> symbolNewEstimate);
+    }
+
+    private PlanNodeStatsEstimate symbolToLiteralNonEquality(Symbol symbol, double literal)
+    {
+        SymbolStatsEstimate symbolStats = inputStatistics.getSymbolStatistics(symbol);
+
+        StatisticRange range = new StatisticRange(symbolStats.getLowValue(), symbolStats.getHighValue(), symbolStats.getDistinctValuesCount());
+        StatisticRange intersectRange = range.intersect(new StatisticRange(literal, literal, 1));
```

Comment:
you could reuse above methods if you instead of `a != C` do `a < C AND a > C`. In this case you would need some chaining (to do `AND) so the first comment in the file applies)